src/include/index/index_group.h

```diff
@@ -359,8 +362,19 @@ class base_index_group {
     }
 
     metadata_.clear_history(timestamp);
-    tiledb::Array::delete_fragments(cached_ctx_, ids_uri(), 0, timestamp);
-    static_cast<group_type*>(this)->clear_history_impl(timestamp);
+    try {
+      tiledb::Array::delete_fragments(cached_ctx_, ids_uri(), 0, timestamp);
+      static_cast<group_type*>(this)->clear_history_impl(timestamp);
+    } catch (const tiledb::TileDBError& e) {
+      if (std::string(e.what()).find("simultaneous open or close operations") !=
```

Comment:
Do we actually keep the arrays open to avoid reopening them across many queries? Can we make sure that we close all arrays consistently especially for library function calls?

Reply:
So the issue is with `partitioned_pq_vectors_`, which is a  `tdbColMajorPartitionedMatrix`:
1. `query_finite_ram()`: here it looks like we actually have a bug because I don't see us clear `partitioned_pq_vectors_` after the query, which we should do so that we can query a second time.
2. `query_infinite_ram()`: here I don't think we want to reset `partitioned_pq_vectors_` because we've just loaded it, and by keeping it in memory we can make subsequent queries faster. Though of course that is at the cost of increased memory usage before that happens, but it seems worth it to me. Instead we could either a. Update `tdbColMajorPartitionedMatrix` to automatically close Array's when it has read all the data or b. Add a `tdbColMajorPartitionedMatrix::markFinishedLoading()` method. I'll plan to merge this, implement one of these in a follow-up (will start with a.), and remove the checks I added here.

Reply:
`query_infinite_ram` should have loaded everything in memory so there is no need to keep the arrays open. Finite ram queries can benefit from not re-opening the array but we don't seem to have that optimization yet.

Reply:
> query_infinite_ram should have loaded everything in memory so there is no need to keep the arrays open.
Yep, exactly. a. vs b. above are just how to implement that in `tdbColMajorPartitionedMatrix` - I'm trying with a. which will have `tdbColMajorPartitionedMatrix` automatically manage this.

> Finite ram queries can benefit from not re-opening the array but we don't seem to have that optimization yet.
True, they'd just need to reset internal bookkeeping / data but could avoid the close -> re-open. But not focusing on that here, though could be nice for the future.

Reply:
Okay have a PR here which implements approach a. to close Array's automatically: https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/448